lib/galaxy/tool_util_models/parameters.py

```diff
@@ -940,6 +940,8 @@ def pydantic_template(self, state_representation: StateRepresentationT) -> Dynam
 
     @property
     def request_requires_value(self) -> bool:
+        # these always have a default value I think/hope so this is never really required
```

Comment:
Why is that ? Optional param without default should be valid ?

Reply:
There is a default value set even for optional parameters - I'm not sure if you explicitly pass null to the API what happens in the tools - I wouldn't make any assumptions without test cases.